File: Motorista/backend/api/src/main/java/com/app/api/Entity/Usuario.java
package com.app.api.Entity;


import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;

import java.time.LocalDate;

@Entity
public class Usuario {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String motorista;
    private String placa;
    private String kmSaida;
    private String kmChegada;
    private String destino;
    private LocalDate data;
    private String horaSaida;
    private String horaChegada;
    private String obs;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getMotorista() {
        return motorista;
    }

    public void setMotorista(String motorista) {
        this.motorista = motorista;
    }

    public String getPlaca() {
        return placa;
    }

    public void setPlaca(String placa) {
        this.placa = placa;
    }

    public String getKmSaida() {
        return kmSaida;
    }

    public void setKmSaida(String kmSaida) {
        this.kmSaida = kmSaida;
    }

    public String getKmChegada() {
        return kmChegada;
    }

    public void setKmChegada(String kmChegada) {
        this.kmChegada = kmChegada;
    }

    public String getDestino() {
        return destino;
    }

    public void setDestino(String destino) {
        this.destino = destino;
    }

    public LocalDate getData() {
        return data;
    }

    public void setData(LocalDate data) {
        this.data = data;
    }

    public String getHoraSaida() {
        return horaSaida;
    }

    public void setHoraSaida(String horaSaida) {
        this.horaSaida = horaSaida;
    }

    public String getHoraChegada() {
        return horaChegada;
    }

    public void setHoraChegada(String horaChegada) {
        this.horaChegada = horaChegada;
    }

    public String getObs() {
        return obs;
    }

    public void setObs(String obs) {
        this.obs = obs;
    }
}

File: Motorista/frontend/driverapp/lib/models/usuario.dart
class Usuario {
  final int? id;
  final String motorista;
  final String placa;
  final String kmSaida;
  final String kmChegada;
  final String destino;
  final String data; // yyyy-MM-dd
  final String horaSaida;
  final String horaChegada;
  final String obs;

  Usuario({
    this.id,
    required this.motorista,
    required this.placa,
    required this.kmSaida,
    required this.kmChegada,
    required this.destino,
    required this.data,
    required this.horaSaida,
    required this.horaChegada,
    required this.obs,
  });

  factory Usuario.fromJson(Map<String, dynamic> json) {
    return Usuario(
      id: json['id'],
      motorista: json['motorista'],
      placa: json['placa'],
      kmSaida: json['kmSaida'],
      kmChegada: json['kmChegada'],
      destino: json['destino'],
      data: json['data'],
      horaSaida: json['horaSaida'],
      horaChegada: json['horaChegada'],
      obs: json['obs'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      "id": id,
      "motorista": motorista,
      "placa": placa,
      "kmSaida": kmSaida,
      "kmChegada": kmChegada,
      "destino": destino,
      "data": data,
      "horaSaida": horaSaida,
      "horaChegada": horaChegada,
      "obs": obs,
    };
  }
}
